From 2613e147f886f7a92f33252acab790daaf4722ee Mon Sep 17 00:00:00 2001
From: Mark Ryan <markdryan@rivosinc.com>
Date: Mon, 3 Nov 2025 09:27:22 +0000
Subject: [PATCH 2/3] Skip test_sunny_day

The test

tests.unit._dynamic_stubs_test.DynamicStubTest.test_sunny_day

seems to be hanging on the riscv64 builders.

Use the nuclear option and strip the test out entirely to ensure no
strange import interactions happen.

Upstream-Status: To upstream [ Further investigation needed ]

Signed-off-by: Mark Ryan <markdryan@rivosinc.com>
Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>
---
 src/python/grpcio_tests/tests/tests.json       |  1 -
 .../tests/unit/_dynamic_stubs_test.py          | 18 ------------------
 2 files changed, 19 deletions(-)

diff --git a/src/python/grpcio_tests/tests/tests.json b/src/python/grpcio_tests/tests/tests.json
index 441a0a29fc..5893d11cda 100644
--- a/src/python/grpcio_tests/tests/tests.json
+++ b/src/python/grpcio_tests/tests/tests.json
@@ -60,7 +60,6 @@
   "tests.unit._cython.cygrpc_test.SecureServerSecureClient",
   "tests.unit._cython.cygrpc_test.TypeSmokeTest",
   "tests.unit._dns_resolver_test.DNSResolverTest",
-  "tests.unit._dynamic_stubs_test.DynamicStubTest",
   "tests.unit._empty_message_test.EmptyMessageTest",
   "tests.unit._error_message_encoding_test.ErrorMessageEncodingTest",
   "tests.unit._exit_test.ExitTest",
diff --git a/src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py b/src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py
index 256fc3026a..a9cec86db5 100644
--- a/src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py
+++ b/src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py
@@ -122,24 +122,6 @@ def _test_grpc_tools_unimportable():
         _assert_unimplemented("grpcio-tools")
 
 
-# NOTE(rbellevi): multiprocessing.Process fails to pickle function objects
-# when they do not come from the "__main__" module, so this test passes
-# if run directly on Windows or MacOS, but not if started by the test runner.
-@unittest.skipIf(
-    os.name == "nt" or "darwin" in sys.platform,
-    "Windows and MacOS multiprocessing unsupported",
-)
-class DynamicStubTest(unittest.TestCase):
-    def test_sunny_day(self):
-        _run_in_subprocess(_test_sunny_day)
-
-    def test_well_known_types(self):
-        _run_in_subprocess(_test_well_known_types)
-
-    def test_grpc_tools_unimportable(self):
-        _run_in_subprocess(_test_grpc_tools_unimportable)
-
-
 if __name__ == "__main__":
     logging.basicConfig()
     unittest.main(verbosity=2)
-- 
2.55.0

